### PR TITLE
Date component - Position of date picker

### PR DESCRIPTION
**Pull request by hegou33**
# Description
With the prop positionDatePicker, the user can choose where he wants to display the datePicker.
# TODO
- [ ] Release notes

# Related Issues / Pull Requests

# Screenshots / Gifs
image

image

image

image
# Testing Instructions

### CODE REVIEW

File: src/components/date/date.js [hunk 1]
```diff
@@ -120,12 +120,12 @@ const Date = Input(InputIcon(InputLabel(InputValidation(class Date extends React
...
     /**
-     * Display the date picker component on top of the input
+     * Choose where displayed the date picker
      *
-     * @property showPickerOnTop
+     * @property positionDatePicker
      * @type {Boolean}
      */
-    showPickerOnTop: PropTypes.bool
+    positionDatePicker: PropTypes.string
```
Comment (edleeks87) [inline src/components/date/date.js]: You could set this to `positionDatePicker: PropTypes.oneOf(OptionsHelper.positionDatePicker)` to limit the accepted prop values to the constants you created

File: src/components/date/__spec__.js [hunk 2]
```diff
@@ -401,22 +401,82 @@ describe('Date', () => {
       expect(style.top).toEqual(10);
     });
 
-    describe('showPickerOnTop is true', () => {
-      it('positions the date picker above the input', () => {
+    describe('check positionDatePicker', () => {
```
Comment (edleeks87) [inline src/components/date/__spec__.js]: There is a lot of repeated code here you could replace a lot of it with a `describe.each()' and work out a way of making your `expect` calls.
https://jestjs.io/docs/en/api#describeeachtable-name-fn-timeout

**Comment by sianford** [comment]
Tested in Safari, Chrome, Firefox, IE11 and Edge ❌ Logged some defects. Informed @hegou33 of the issues.

**Comment by sianford** [comment]
Tested in Safari, Chrome, Firefox, IE11 and Edge. Only remaining issue with the bottom of the date input field disappearing when the date picker is in a bottom-right position exists in v42.n of MS Edge only. This does not occur in v44.n, therefore this will not be fixed and will not pose a problem for users providing they update MS Edge to the most recent version available. Testing passed ✅

**Pull request merged**